optuna/visualization/_intermediate_values.py

```diff
@@ -113,12 +123,16 @@ def _get_intermediate_plot(info: _IntermediatePlotInfo) -> "go.Figure":
     if len(trial_infos) == 0:
         return go.Figure(data=[], layout=layout)
 
+    default_marker = {"maxdisplayed": 10}
+
     traces = [
         go.Scatter(
             x=tuple((x for x, _ in tinfo.sorted_intermediate_values)),
             y=tuple((y for _, y in tinfo.sorted_intermediate_values)),
             mode="lines+markers",
-            marker={"maxdisplayed": 10},
+            marker=default_marker
+            if tinfo.feasible
+            else default_marker | {"color": "#CCCCCC"},  # type: ignore[operator]
```

Comment:
Using `|` as a union operator for `dict` is only available since `Python 3.9`. Several ways to do it before `Python 3.9` is mentioned in [PEP584](https://peps.python.org/pep-0584/), so please use one of them.